src/classes/BGE_ConfigurationWizard_CTRL.cls

```diff
@@ -115,44 +115,27 @@ public with sharing class BGE_ConfigurationWizard_CTRL {
 
         Set<String> allowedObjects = new Set<String>{
             'Opportunity',
-            'Payment'
+            'npe01__OppPayment__c'
         };
 
         List<DescribeFieldResult> dfrs = getRelevantFields(allowedObjects);
 
         for (DescribeFieldResult dfr : dfrs) {
 
-            String dataImportFieldMappedObject = BDI_DataImportService.getTargetObject(dfr.getName());
-
             BGEField field = new BGEField();
-
-            field.sObjectName = dataImportFieldMappedObject;
-
-            // this will require iteration if/when we allow objects beyond Opp and Payment
-            String translatedObjectLabel;
-            if (dataImportFieldMappedObject == 'Opportunity') {
-                translatedObjectLabel = UTIL_Describe.getObjectLabel(dataImportFieldMappedObject);
-            } else if (dataImportFieldMappedObject == 'Payment') {
-                translatedObjectLabel = UTIL_Describe.getObjectLabel('npe01__OppPayment__c');
-            }
-            field.sObjectLabel = translatedObjectLabel;
-
+            field.name = dfr.getName();
             // TODO: pull these labels from target fields' schema based on help text mapping
             Boolean fieldHasSpacesAndObjName = dfr.getLabel().contains(' ') && dfr.getLabel().split(' ',2)[0] == field.sObjectName;
             field.label = (fieldHasSpacesAndObjName) ? dfr.getLabel().split(' ',2)[1] : dfr.getLabel();
-            field.name = dfr.getName();
 
-            String unnamespacedField = UTIL_Namespace.StrTokenRemoveNSPrefix(field.name);
-            if (conditionallyRequiredFields.contains(unnamespacedField)) {
+            field.sObjectName = BDI_DataImportService.getTargetObject(dfr.getName());
+            field.sObjectLabel = UTIL_Describe.getObjectLabel(field.sObjectName);
+
+            if (conditionallyRequiredFields.contains(field.name)) {
                 field.conditionallyRequired = true;
             }
 
-            String rawFieldType = dfr.getType().name().toLowerCase();
             field.type = UTIL_LightningComponentSupport.getLightningInputDataType(dfr.getType());
```

Comment:
Do we need type in this context?

Reply:
Yep, still do! It's used to see if it's a reference and we have to do some converting in 2 places in the JS, so I'd vote to keep this for now, or make this a separate ticket. Maybe part of a ticket to increase the speed of the load, since it seems to have gotten slow again?